spec/controllers/teams_controller_spec.rb

```diff
@@ -113,5 +121,4 @@
       expect(response).to redirect_to(teams_url)
     end
   end
-
```

Comment:
Consider adding tests for the newly created controller actions/routes `assign_ownership` and `delete_ownership`. Have look at other tests in this file. An idea:
`describe "POST #delete_ownership" do
    it "deletes a ownership of a team member" do
      team = FactoryBot.create :team
      user = FactoryBot.create :user
      another_user = FactoryBot.create :user

      team.owners << user
      team.owners << another_user
      team.members << user
      team.members << another_user

      expect {
        post :delete_ownership, params: { team_owner: another_user.id }
      }.to change(TeamOwner, :count).by(-1)
       .and change(TeamMember, :count).by(0)
    end

    it "redirects to the teams list" do
      team = FactoryBot.create :team
      user = FactoryBot.create :user
      another_user = FactoryBot.create :user

      team.owners << user
      team.owners << another_user
      team.members << user
      team.members << another_user

      post :delete_ownership, params: { team_owner: another_user.id }
      expect(response).to redirect_to(teams_url)
    end
  end`

Reply:
A problem I detected: Objects created by a factory do not have an ID.

Reply:
oh fanks! didn't know that -- might help me a great deal for those f***ed up controller-tests with CanCan -- which, btw, are not in themselves best practice and are often recommended not to be written at all for there are cumbersome pieces of code. but whatever test coverage requirements dictate must.be.obeyed.